docs/guide/newReplacement.md

```diff
@@ -0,0 +1,337 @@
+  */
+ replace(
+         node: NodePath, // AST Node wrapper which can be used to navigate to the parent
+         name: string, // variable name
+         fnName: string,
+         oldModuleCall: string,
+         config: {newVariableName: string} //config object
+ ): void {
+
+  // find calls e.g. to `myObject.doCalculateIt(7)`
+  if (
+          // is a call with positive integer number as argument
+          node.value.type === Syntax.CallExpression && node.value.arguments.length === 1
+          && node.value.arguments[0].type === Syntax.Literal
+          && typeof node.value.arguments[0].value === "number"
+          && node.value.callee.type === Syntax.MemberExpression
+  ) {
+   node.value.callee.object = builders.identifier(name);
+   // even
+   if (node.value.arguments[0].value % 2 === 0)  {
+    node.value.callee.property = builders.identifier("even");
+    // odd
+   } else {
+    node.value.callee.property = builders.identifier("odd");
+   }
+  }
+ }
+}
+
+module.exports = new MyCustomReplacer();
+```
+
+#### Extender
+* **How2**: there should be an import added for module "a"
+```ts
+import {Extender} from "../../../dependencies";
+import {SapUiDefineCall} from "../../../util/SapUiDefineCall";
+
+/**
+ * Adds an import to the define statement
+ */
+class MyCustomExtender implements Extender {
+	extend(
+		defineCall: SapUiDefineCall,
+		config: {
+			newModulePath: string;
+			newVariableName: string;
+		}
+	): boolean {
+		return defineCall.addDependency(
+			config.newModulePath,
+			config.newVariableName
+		);
+	}
+}
+
+module.exports = new MyCustomExtender();
+```
+
+##### How to test it
+Create 3 files inside the folder test/addMissingDependencies
+* source file, e.g. mytestfile.js
+* expected file, e.g. mytestfile.expected.js
+* config file, e.g. mytestfile.config.json
+
+Reference the test file inside `test/addMissingDependencies.ts`
+
+```ts
+it("new test for mytest", done => {
+    const subDir = rootDir + "replaceJQuery2/";
+    const expectedContent = fs.readFileSync(
+        subDir + "mytestfile.expected.js",
+        "utf8"
+    );
+    const config = JSON.parse(
+        fs.readFileSync(subDir + "mytestfile.config.json", "utf8")
+    );
+    const module = new CustomFileInfo(subDir + "mytestfile.js");
+    analyseMigrateAndTest(
+        module,
+        true,
+        expectedContent,
+        config,
+        done,
+        []
+    );
+});
+```
+
+and the build it and run the unit test
+
+Note: use `npm run build` to compile the typescript sources
+
+Run `npm run unit` to run the unit tests
+
+Use `it.only(...)` in test files, such that only a [single test](https://mochajs.org/#exclusive-tests) is run.
+
+## Different replacements depending on version
+If a code modification depends on the UI5 version being used the following approach should be taken:
+
+The call to replace (second level) should have an `@version` attached and the object should have the property
```

Comment:
“call to replace” -> "**name of the config**"

Reply:
done